src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -104,8 +105,13 @@ public void End()
                 await _outputConsole.WriteLineAsync(Resources.Finished);
                 await _outputConsole.WriteLineAsync(string.Empty);
 
-                // Give the error list focus
-                await _errorList.Value.BringToFrontIfSettingsPermitAsync();
+                if (_errorsReported)
+                {
+                    // Give the error list focus
+                    await _errorList.Value.BringToFrontIfSettingsPermitAsync();
```

Comment:
Seems like all places BringToFrontIfSettingsPermitAsync is called actually have a check for whether there's any errors added by NuGet before showing the error list.

Is it possible to potentially move that logic into the error list and maybe repurpose it as `BringToFrontIfNeeded` or soemthing like that?

That would avoid future issues like this where we bring focus to the error list while not having any errors.

Reply:
The only other usage I'm seeing is Restore, which appears to show the error list after restore completes: 

https://github.com/NuGet/NuGet.Client/blob/181b65dad9f440c7a31fe673abc59c258f224ada/src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs#L125

I think you're saying refactor the ErrorListDataSource and restore? I'm happy to that in another PR, but prefer not to touch restore in this PR in case a regression is introduced.

Reply:
I understand - just not sure it's appropriate to bring more risk into this PR which is largely not related to Restore.

Reply:
The way I was thinking about this PR is that it's fixing up unnecessary error list behavior, so I thought tweaking that might be in scope.

I think we should proactively address potential technical debt as we see it, and this looks like it to me.

That being said, that's just a perspective and not a hardline for this PR (I approved it). 

Your call whether you want to resolve it here or leave as a future thing.